### PR TITLE
Integrate code from get_error.py

### PR DESCRIPTION
**Pull request by yashika51**
@Acrylami I have tried to integrate your code with mine.
Also as suggested the script now prints the answers as well. 
Can you please test if this works on your machine, the errors if any would be due to permissions or a file access problem?
No rush, whenever you're available, please let me know.

Meanwhile @Levy-Naibei, please check as well whenever possible. 

Thanks both!

### CODE REVIEW

**Comment by Acrylami** [comment]
Hi Yashika I only see one changed file here and it doesn't look like it integrates it with any other files. I tried running cli.py and get_error.py and couldn't see any answers either. It looks like there might have been more changes you made that haven't been committed to this branch properly?

**Comment by yashika51** [comment]
@Acrylami oh I meant I combined/integrated code not files. I want you to check if `search_answers.py` is working fine. 
To test this, we have to pass `file_name` to the object of `class ask`. Then just call the function `get_answer()` with the help of object.

**Comment by yashika51** [comment]
For integrating files, we need to discuss it with @Levy-Naibei, how the output would be displayed in cli. The `search()` function would probably get the output from the `search_answers.py` and then it's up to @Levy-Naibei how to display it.

**Review by levynaibei — CHANGES_REQUESTED** [review]
Hey @yashika51 ,
Where has "items" been declared?

**Comment by yashika51** [comment]
@Levy-Naibei `items` is a key in the JSON file we get from the API.

**Comment by levynaibei** [comment]
> @Levy-Naibei `items` is a key in the JSON file we get from the API.

**Comment by levynaibei** [comment]
> @Levy-Naibei `items` is a key in the JSON file we get from the API.

there is a key error in it, let me see that json file ..

**Comment by levynaibei** [comment]
> > @Levy-Naibei `items` is a key in the JSON file we get from the API.

@yashika51 , 
where is the link to that json file?

**Comment by yashika51** [comment]
@Levy-Naibei you have to read the API docs, the json file is returned when we run the query and it depends on what the error is. 
The error should not be with items or code in `search_answers.py`. Its returning the answers properly. You have to check `cli.py` for integration issues instead.

**Comment by levynaibei** [comment]
> @Levy-Naibei you have to read the API docs, the json file is returned when we run the query and it depends on what the error is.
> The error should not be with items or code in `search_answers.py`. Its returning the answers properly. You have to check `cli.py` for integration issues instead.

 its just that i am keyerror. ok  let me check cli.py

**Comment by levynaibei** [comment]
Hey @yashika51 
the import  and use of class `ask` is in cli.py is ok, the error we are getting should be in `items`. please check the data structure well in the API, may be `items` is a list of dictionary yet you are iterating it like a plain list.. plis check that first and let me know

**Comment by yashika51** [comment]
Levy running search_answers.py would tell you there's no error at all.
I have checked that part already. It's something with integration.
Let me know if search_answers.py gives any error. The instructions to test
is given in one of the previous PRs including screenshot. Thanks.

**Comment by levynaibei** [comment]
which error message are you getting ?

**Comment by levynaibei** [comment]
@yashika51 , I am debugging using pdb and i can still see the error lies in data['items']  .. so just share that data i have a look at it

**Comment by levynaibei** [comment]
Hey @Acrylami and @yashika51, after long debugging, i have found out a fix that will make the CLI work.

Our error message(relevant info) needs to be cleaned up to `self.error_message='unsupported operand type(s) for int and str'`
if i hard code `unsupported operand type(s) for int and str'` in the class initialization method and run `python cli.py search "python error_test.py"` on the terminal, the application works well - it is able to fetch answers from stackoverflow.

currently it has a lot of irrelevant info `'Traceback (most recent call last)', '\n  File "error_test.py", line 4, in <module>\n    a = 10 + "b"\nTypeError', ' unsupported operand type(s) for +', " 'int' and 'str'"]'` .After showing this, you notice the cli breaks

so the `error_message` in `get_error.py` module needs to be cleansed to look like one above - only error message needed and remove info such as Traceback, File, Typerror etc

Hope this helps.. there you go

**Comment by Acrylami** [comment]
Hi Levy thanks for this
What are the steps for reproducing this error?
The last time I ran the code is did not have all this irrelevant
information, as can be seen in a previous PR screenshot or in our discord
chat
It used to just say "type error", "unsupported operand....", "Int +
string..." In the list

File: search_answers.py [hunk 1]
```diff
@@ -12,14 +12,16 @@
 import urllib.request
 import json
 from bs4 import BeautifulSoup
-import get_error
+from get_error import main
```
Comment (Acrylami) [inline search_answers.py:15]: Should not be needed to import get_error anymore

File: cli.py [hunk 2]
```diff
@@ -34,6 +35,9 @@ def search(query):
         #Print all relevant search info
         print("Relevant information for search: ")
         print(relevant_info)
+        a=ask(relevant_info)
+        answers=a.get_answer()
+        print(answers[0])
```
Comment (Acrylami) [inline cli.py:40]: I've changed it to only print the first answer to make it easier to look at

**Review by Acrylami — APPROVED** [review]
Works for me.
Type "python cli.py search "python error_test.py"" to run
In future more work will be needed:

1. With the API to only return the most upvoted answer the the SO question- it seems it is returning all answers even bad answers? @yashika51 
2. With the user interface to display the answer in a readable way like issue #6 @Levy-Naibei

**Pull request merged**